dali/kernels/test/kernel_poc_test.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2018-2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <random>
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/test/test_tensors.h"
+#include "dali/kernels/tensor_shape_str.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+
+using std::cout;
+using std::cerr;
+using std::endl;
+
+namespace dali {
+namespace kernels {
+
+template <typename Input1, typename Input2, typename Output>
+struct MADKernel {
+  static KernelRequirements GetRequirements(
+      KernelContext &context,
+      const InListCPU<Input1, 3> &i1,
+      const InListCPU<Input2, 3> &i2,
+      float A) {
+    KernelRequirements req;
+    req.output_shapes = { i1.shape };
+    return req;
+  }
+
+  static void Run(
+      KernelContext &context,
+      const OutListCPU<Output, 3> &o,
```

Comment:
I guess we could come back to discussion about CPU kernels supporting TensorList or Tensors?
Was there any definite decision, cause I don't remember. How it will work with the executor? This currently adds one dimension to each implementation and will be awkward to use if we want to write adapters like in case of sequences.

Reply:
Yeah, I think we should treat TensorsView as inputs. Even if we decide to process tensor lists in Operators, it might still be valuable to have sample-wise kernels, especially for CPU.

Reply:
Changed.